File: springCurseWork/src/main/java/com/example/springcursework/servise/AttendanceService.java
package com.example.springcursework.servise;

import com.example.springcursework.model.Attendance;
import com.example.springcursework.model.AttendanceView;

import java.util.List;

public interface AttendanceService {
    public Attendance insert(Attendance attendanceVO);

    public List<Attendance> findAll();

    public void delete(int id);

    public Attendance findById(int id);

    public Attendance updateAttendance(int id, Attendance attendanceVO);

    List<AttendanceView> findByGroupScheduleId(int groupScheduleId);

    List<AttendanceView> updateAttendances(int groupScheduleId, List<AttendanceView> attendances);
}

File: SchoolAngular/src/app/service/attendance.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import {Attendance} from "../model/attendance.model";
import {AttendanceView} from "../model/attendance.view.model";

@Injectable()
export class AttendanceService {

  private basUrl = "http://localhost:8080/attendance"

  constructor(private httpClient: HttpClient) {

  }

  getAttendanceList(): Observable<Attendance[]> {
    return this.httpClient.get<Attendance[]>(`${this.basUrl}`);
  }

  createAttendance(attendance: Attendance): Observable<Attendance> {
    return this.httpClient.post<Attendance>(`${this.basUrl}`, attendance);
  }

  getAttendanceById(id: number): Observable<Attendance>{
    return this.httpClient.get<Attendance>(`${this.basUrl}/${id}`);
  }

  updateAttendance(id:number, attendance:Attendance): Observable<Object>{
    return this.httpClient.put(`${this.basUrl}/${id}`, attendance);
  }

  deleteAttendance(id:number): Observable<Object>{
    return this.httpClient.delete(`${this.basUrl}/${id}`);
  }

  findForGroupSchedule(groupScheduleId: number): Observable<AttendanceView[]> {
    return this.httpClient.get<AttendanceView[]>(`${this.basUrl}/find-for-group-schedule/${groupScheduleId}`);
  }

  updateAttendances(groupScheduleId: number, attendance: AttendanceView[]): Observable<AttendanceView[]>{
    return this.httpClient.put<AttendanceView[]>(`${this.basUrl}/update-attendances/${groupScheduleId}`, attendance);
  }

}
